contract-tests/test_v3_api.py

```
@@ -0,0 +1,240 @@
+        return json.loads(schema_file.read())
+
+
+def assert_valid_schema(data, schema_file):
+    schema = _load_json_schema(schema_file)
+    return jsonschema.validate(data, schema)
+
+
+def test_v3_actions(conf, requests_session):
+    # Get a random action and validate it
+    r = requests_session.get(conf.getoption("server") + "/api/v3/action/")
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0:
+        pytest.skip("No list of v3 actions found")
+
+    key = randint(0, len(data["results"]) - 1)
+    action_id = data["results"][key]["id"]
+
+    r = requests_session.get(conf.getoption("server") + "/api/v3/action/{}/".format(action_id))
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0:
+        pytest.skip("Could not find a v3 action with the ID {}".format(action_id))
+
+    assert r.status_code == 200
+    assert_valid_schema(data, "v3.action.schema")
+
+
+def test_v3_approval_request(conf, requests_session):
+    r = requests_session.get(conf.getoption("server") + "/api/v3/approval_request/")
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0 or len(data["results"]) == 0:
+        pytest.skip("No list of v3 approval requests found")
+
+    assert r.status_code == 200
+
+
+def test_v3_approval_request_with_id(conf, requests_session):
+    # Get a random approval request and validate it
+    r = requests_session.get(conf.getoption("server") + "/api/v3/approval_request/")
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0 or len(data["results"]) == 0:
+        pytest.skip("Could not find a list of v3 approval requests")
+
+    key = randint(0, len(data["results"]) - 1)
+    approval_request_id = data["results"][key]["id"]
+
+    r = requests_session.get(
+        conf.getoption("server") + "/api/v3/approval_request/{}/".format(approval_request_id)
+    )
+    r.raise_for_status()
+    data = r.json()
+
+    if len(data) == 0:
+        pytest.skip(

```

I think this should also fail for having a missing detail page.